### DISCUSSION TITLE
Did `Html` became required in 9.5.3 in custom _document?

### DISCUSSION THREAD

**Discussion by glenngijsberts** [vercel/next.js#16908] [Help] [answered]
This morning we updated our next version from `9.5.2` to `9.5.3`. After a while we noticed a warning in our dev console saying:

```
Expected Document Component Html was not rendered. Make sure you render them in your custom `_document`
See more info here https://err.sh/next.js/missing-document-component
```

After digging I found out that we didn't use the `Html` tag, however, we haven't use that tag since the startup of our project (~2 years ago). Did it became required from 9.5.3? We haven't seen that warning before. Nothing was mentioned in the update logs (from dependabot).

Thanks in advance 👍

**Comment by timneutkens** [comment, marked as answer]
We added the warning recently as we saw people did not include it and in order for all features to work correctly (also looking to the future). E.g. the `Html` component is needed for AMP to work correctly.

**Reply by Vadorequest** [reply]
Good move, I wasn't aware of this, and had been using the normal `html` tag for the past 2 years too.

**Reply by glenngijsberts** [reply]
Thanks! 

I just realised it's literally saying that the `Html` component is missing, I thought just _some_ Html was missing. My bad 😅

**Reply by Vadorequest** [reply]
Yeah, I got confused at first too.

Basically, it should become:
```tsx
import Document, { DocumentContext, DocumentProps, Head, Main, NextScript, Html } from 'next/document';

...

<Html lang={lang}>
        <Head />
        <body
          className={classnames(
            // XXX Those variables are added to grant more flexibility if ever needed. They're not used at the moment
            'nrn', // All styles are bound to this, if you remove/rename, it'll break all CSS in src/components/appBootstrap/UniversalGlobalStyles.tsx
            `${process.env.NEXT_PUBLIC_APP_NAME}`, // From package.json:name

            // Localisation-based styles are very useful (e.g: resize text based on locale or language)
            `locale-${locale}`,
            `lang-${lang}`,

            // For customer/stage/version based styles, could be handy in rare cases
            `${process.env.NEXT_PUBLIC_CUSTOMER_REF}`,
            `stage-${process.env.NEXT_PUBLIC_APP_STAGE}`,
            `v${process.env.NEXT_PUBLIC_APP_VERSION}`, // From package.json:version
          )}
        >
          <Main />
          <NextScript />
        </body>
      </Html>
```

Instead of 

```tsx
<html lang={lang}>
        <Head />
        <body
          className={classnames(
            // XXX Those variables are added to grant more flexibility if ever needed. They're not used at the moment
            'nrn', // All styles are bound to this, if you remove/rename, it'll break all CSS in src/components/appBootstrap/UniversalGlobalStyles.tsx
            `${process.env.NEXT_PUBLIC_APP_NAME}`, // From package.json:name

            // Localisation-based styles are very useful (e.g: resize text based on locale or language)
            `locale-${locale}`,
            `lang-${lang}`,

            // For customer/stage/version based styles, could be handy in rare cases
            `${process.env.NEXT_PUBLIC_CUSTOMER_REF}`,
            `stage-${process.env.NEXT_PUBLIC_APP_STAGE}`,
            `v${process.env.NEXT_PUBLIC_APP_VERSION}`, // From package.json:version
          )}
        >
          <Main />
          <NextScript />
        </body>
      </html>
```

Full code: https://github.com/UnlyEd/next-right-now/blob/17f17d965b6ecd18f7a99d903fec9cde56bc18b3/src/pages/_document.tsx#L17

**Reply by inesmaio** [reply]
Thanks a lot!!!

**Comment by Vadorequest** [comment]
Same thing today, I switch from `html` to `Html`.

It won't cause an issue with the project, but I don't know what it does. _Well, [Tim explained it](https://github.com/vercel/next.js/discussions/16908#discussioncomment-63638)_

Here is the TS signature:
```ts
export declare function Html(props: React.DetailedHTMLProps<React.HtmlHTMLAttributes<HTMLHtmlElement>, HTMLHtmlElement>): JSX.Element;

```

Either way, doesn't seem like a regression nor a non-backward compatible change. I didn't encounter any issue by switching to `Html`, I believe it's just a warning to benefit from better TS typings, maybe. (or another good reason)

See https://github.com/UnlyEd/next-right-now/pull/161
